scripts/legacysurvey/build_parent_sample.py

```diff
@@ -1,39 +1,174 @@
+
+    def get_object_mask(self) -> np.ndarray:
+        mask = np.zeros(self.cutout.shape).astype(np.uint8)
+        # Get catalob object pixel coordinates in cutout
+        object_coordinates = SkyCoord(ra=self.catalog["RA"], dec=self.catalog["DEC"])
+        i, j = self.cutout.wcs.world_to_array_index(object_coordinates)
+        centers = np.stack([j, i]).T
+        radii = self.catalog["SHAPE_R"].value / ARCSEC_PER_PIXEL
+        object_types = self.catalog["TYPE"].value.astype(str)
+        for c, r, t in zip(centers, radii, object_types):
+            # Enforce a minimum size of the disk mask
+            r = max(2, r)
+            rr, cc = skimage.draw.disk(c, r, shape=self.cutout.shape)
+            mask[cc, rr] = OBJECT_TYPE_COLOR[t]
+        return mask
+
+    def get_brightest_object_catalog(
+        self, n_objects: int = 20
+    ) -> Dict[str, List[float]]:
+        positive_flux_indices = self.catalog["FLUX_Z"] > 0
+        magnitude = np.zeros_like(self.catalog["FLUX_Z"].value)
+        magnitude[positive_flux_indices] = 22.5 - 2.5 * np.log10(
+            self.catalog["FLUX_Z"].value[positive_flux_indices]
+            / self.catalog["MW_TRANSMISSION_Z"].value[positive_flux_indices]
+        )
+        self.catalog["MAG_Z"] = magnitude
+        self.catalog.sort(keys="MAG_Z", reverse=True)
+        brightest_object_data = {key: [] for key in NEARBY_CATALOG_INFORMATION}
+        brightest_object_catalog = self.catalog[:n_objects]
+        # Check there is at least one object
+        # The center object should at least be in the catalog
+        assert (
+            len(brightest_object_catalog) > 0
+        ), "The nearby catalog should at least contain one object."
+
+        for obj in brightest_object_catalog:
+            for key in NEARBY_CATALOG_INFORMATION:
+                if key == "TYPE":
+                    data = OBJECT_TYPE_COLOR[obj[key]]
+                else:
+                    data = obj[key]
+                brightest_object_data[key].append(data)
+
+        # Pad with zeros data if necessary
+        for _ in range(len(brightest_object_catalog), n_objects):
+            for key in NEARBY_CATALOG_INFORMATION:
+                brightest_object_data[key].append(0)
+
+        return brightest_object_data
+
+
+def print_healpix_error(err, healpix_filename: str):
+    print(f"Failed to write {healpix_filename} due to {err}")
+
+
+def select_observations(catalog, zmag_cut=21.0) -> List[bool]:
+    """Selection function applied to retrieve relevant observation from the DECaLS DR10 South catalog.
+    Observation are deemed relevant based on minimum magnitude, availability in all bands and bit masks.
     """
     # Magnitude cut
-    mask_mag = (22.5 - 2.5*np.log10(catalog['FLUX_Z']/catalog['MW_TRANSMISSION_Z'])) < zmag_cut
+    magnitude = np.zeros_like(catalog["FLUX_Z"])
+    positive_flux_indices = catalog["FLUX_Z"] > 0.0
+    magnitude[positive_flux_indices] = 22.5 - 2.5 * np.log10(
```

Comment:
I reverted back to the previous code

Reply:
Good catch, I think I had implemented this because of the complaints on the negative values passed to the log. But I didn't realize it would impact the mask for negative values of the flux.